### PR TITLE
Pm mypackages - update package version

### PR DESCRIPTION
**Pull request by dnenov**
### Purpose

A final edition to the package manager updates focusing on publishing a new version of the existing package. Based on https://github.com/DynamoDS/Dynamo/pull/14645.

- only the creator of the package can publish a new version of the package
- the feature is accessible once the package is installed locally through either;
- - the `installed packages` tab
- - `my packages` tab where the 'Install' button will turn into a 'publish version' button
- when activated, the user will be directed to the 'publish package' tab with the details of the package pre-filled in
- the user won't be able to change the name of the package
- the contents of the package will be automatically populated for the user, but the user can make changes

#### UI Changes

update package version

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [x] This PR modifies some build requirements and the readme is updated
- [x] This PR contains no files larger than 50 MB

### Release Notes

- publish package version changes

### Reviewers

@QilongTang 
@reddyashish 

### FYIs

@Amoursol

### CODE REVIEW

**Comment by Amoursol** [comment]
@dnenov Installed Packages text still dramatically smaller than the rest of this UI - is this addressed in a different PR?

File: src/DynamoCoreWpf/Properties/Resources.en-US.resx [hunk 1]
```diff
@@ -3612,6 +3612,9 @@ You can manage this in Preferences -&gt; Security.</value>
   <data name="PackageDeprecatedTooltip" xml:space="preserve">
     <value>This package is outdated and cannot be installed.</value>
   </data>
+  <data name="PackageManagerMyPackagesPublishVersion" xml:space="preserve">
```
Comment (reddyashish) [inline src/DynamoCoreWpf/Properties/Resources.en-US.resx:3615]: Add this resource to the general resources file too.

File: src/DynamoCoreWpf/Views/PackageManager/Controls/PackageManagerPublishControl.xaml.cs [hunk 2]
```diff
@@ -32,17 +32,43 @@ public PackageManagerPublishControl()
             InitializeComponent();
 
             this.Loaded += InitializeContext;
+            this.DataContextChanged += ThisDataContextChanged;
```
Comment (reddyashish) [inline src/DynamoCoreWpf/Views/PackageManager/Controls/PackageManagerPublishControl.xaml.cs:35]: Following the pattern in our code base, can we change the event name to OnDataContextChanged or PackageManagerPublishControl_DataContextChanged

**Review by reddyashish — COMMENTED** [review]
LGTM with one comment.
Running the self-serve job here: https://master-15.jenkins.autodesk.com/view/DYN/job/DYN-DevCI_Self_Service/1311/

**Pull request merged**